File: 3419-maximum-points-inside-the-square/maximum-points-inside-the-square.java
class Solution {
    public int maxPointsInsideSquare(int[][] points, String s) {
        TreeMap<Integer, StringBuilder> mpp = new TreeMap<>();
        int ma = 0;

        for (int i = 0; i < points.length; ++i) {
            int x = Math.abs(points[i][0]);
            int y = Math.abs(points[i][1]);
            int dist = Math.max(x, y);
            mpp.putIfAbsent(dist, new StringBuilder());
            mpp.get(dist).append(s.charAt(i));
        }

        int[] count = new int[26];
        for (Map.Entry<Integer, StringBuilder> entry : mpp.entrySet()) {
            boolean flag = true;
            for (char ch : entry.getValue().toString().toCharArray()) {
                count[ch - 'a']++;
                if (count[ch - 'a'] > 1) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                ma += entry.getValue().length();
            } else break;
        }
        return ma;
    }
}

File: 3419-maximum-points-inside-the-square/maximum-points-inside-the-square.cpp

class Solution {
public:
    int maxPointsInsideSquare(vector<vector<int>>& points, string s) {
        map<int, string> mpp;  
        int ma = 0;
        for (int i = 0; i < points.size(); ++i) {
            int x = abs(points[i][0]);
            int y = abs(points[i][1]);
            int dist = max(x, y); 
            mpp[dist] += s[i];
        }
        vector<int> count(26, 0);
        for (auto& it : mpp) {            
            bool flag = true;
            for (char ch : it.second) {
                count[ch - 'a']++;
                if (count[ch - 'a'] > 1) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                ma +=(it.second.length());
            }
            else break;
        }

        return ma;
    }
};